app/controllers/tracks_controller.rb

```diff
@@ -0,0 +1,21 @@
+class TracksController < ApplicationController
+  before_action :use_track, except: :index
+
+  def authenticated_index
+    @tracks = Track.all
+  end
+
+  def authenticated_show
+    @user_track = current_user.user_tracks.find_by(track: @track)
+  end
+
+  def join
+    UserTrack::Create.(current_user, @track)
+    redirect_to action: :show
+  end
+
+  private
+  def use_track
+    @track = Track.find(params[:id])
```

Comment:
```suggestion
    @track = Track.find(params[:id])
```

I personally prefer to remove the magic and doing `Track.friendly.find` or whatever the syntax is.

Reply:
I'm sensing a theme in your feelings about Ruby magic here 😉

Reply:
Less Ruby magic is also easier for me to read as a Ruby n00b 😉

Reply:
I love Ruby magic, but working with non ruby seniors has learned me to embrace the features whilst still being explicit :)

Reply:
So, I agree. I'm also going to put out there the reminder for the four of us as this process continues that we don't accept PRs to this repo, and that juniors aren't getting near it 🙂 

But the principle of least surpirses is something I'm passionate about.